src/main/kotlin/com/github/imflog/schema/registry/parser/AvroSchemaParser.kt

```diff
@@ -17,8 +22,45 @@ class AvroSchemaParser(
         schemaContent: String,
         localReferences: List<LocalReference>
     ): String {
-        val parser = Schema.Parser()
-        localReferences.map { reference -> reference.content(rootDir) }.reversed().forEach { parser.parse(it) }
-        return parser.parse(schemaContent).toString()
+        val jsonObj = JSONObject(schemaContent)
+        if(jsonObj.has(FIELDS)) {
+            jsonObj.put(FIELDS,
+                JSONArray(jsonObj
+                    .getJSONArray(FIELDS)
+                    .toMutableList()
+                    .mapInPlace { if (it is JSONObject) replaceLocalReference(it, localReferences) else it }
```

Comment:
Is the mutability really required ? I think that we could `map` immutably and that would work. WDYT ?

Reply:
Yes you right, I don't know why I went with a map in place, when a simple would do the trick